internal/serverless/client.go

```diff
@@ -0,0 +1,332 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package serverless
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const (
+	defaultHostURL = "https://cloud.elastic.co"
+
+	projectsAPI = "/api/v1/serverless/projects"
+)
+
+type Client struct {
+	host   string
+	apiKey string
+}
+
+// ClientOption is functional option modifying Serverless API client.
+type ClientOption func(*Client)
+
+var (
+	elasticCloudApiKeyEnv   = "EC_API_KEY"
+	elasticCloudEndpointEnv = "EC_HOST"
+
+	ErrProjectNotExist = errors.New("project does not exist")
+)
+
+func NewClient(opts ...ClientOption) (*Client, error) {
+	apiKey := os.Getenv(elasticCloudApiKeyEnv)
+	if apiKey == "" {
+		return nil, fmt.Errorf("unable to obtain value from %s environment variable", elasticCloudApiKeyEnv)
+	}
+	c := &Client{
+		host:   defaultHostURL,
+		apiKey: apiKey,
+	}
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	host := os.Getenv(elasticCloudEndpointEnv)
+	if host != "" {
+		c.host = host
+	}
```

Comment:
If there is an environment variable defined it would have preference over the configuration set in the profile.